src/randoop/main/GenTests.java

```diff
@@ -429,264 +399,335 @@ public boolean handle(String[] args) throws RandoopTextuiException {
-          non_timeout_seqs.add (es);
-        // This test suggests a shorter way to implement this method.
-        assert keep == !es.throwsException(TimeoutExceededException.class);
       }
-      sequences = non_timeout_seqs;
     }
-
-    // If specified, remove any sequences that don't include the target class
-    // System.out.printf ("test_classes regex = %s%n",
-    //                   GenInputsAbstract.test_classes);
-    if (GenInputsAbstract.test_classes != null) {
-      List<ExecutableSequence> tc_seqs = new ArrayList<ExecutableSequence>();
-      for (ExecutableSequence es : sequences) {
-        boolean keep = false;
-        for (Variable v : es.sequence.getAllVariables()) {
-          if (GenInputsAbstract.test_classes.matcher (v.getType().getName())
-              .matches()) {
-            keep = true;
-            break;
-          }
+    
+    if (! GenInputsAbstract.no_regression_tests) {
+      List<ExecutableSequence> regressionSequences = explorer.getRegressionSequences();
+      if (regressionSequences.size() > 0) {
+        if (! GenInputsAbstract.noprogressdisplay) {
+          System.out.format("%nRegression test output:%n");
+        }
+        outputTests(regressionSequences, GenInputsAbstract.regression_test_filename);
+      } else {
+        if (! GenInputsAbstract.noprogressdisplay) {
+          System.out.format("No regression tests to output%n");
         }
-        if (keep)
-          tc_seqs.add (es);
       }
-      sequences = tc_seqs;
-      System.out.printf ("%n%d sequences include %s%n", sequences.size(),
-                         GenInputsAbstract.test_classes);
     }
+    
+    return true;
+  }
 
-    // If specified remove any sequences that are used as inputs in other
-    // tests.  These sequences are redundant.
-    //
-    // While we're at it, remove the useless sequence "new Object()".
-    Sequence newObj = new Sequence().extend(objectConstructor);
-    if (GenInputsAbstract.remove_subsequences) {
-      List<ExecutableSequence> unique_seqs
-        = new ArrayList<ExecutableSequence>();
-      Set<Sequence> subsumed_seqs = explorer.subsumed_sequences();
-      for (ExecutableSequence es : sequences) {
-        if (!subsumed_seqs.contains(es.sequence) && !es.sequence.equals(newObj)) {
-          unique_seqs.add(es);
-        }
+  /**
+   * Builds the test predicate that determines whether a particular sequence will
+   * be included in the output based on command-line arguments.
+   * 
+   * @param objectConstructor  the constructor for the Object class
+   * @return the predicate 
+   */
+  private Predicate<ExecutableSequence> createTestOutputPredicate(ConstructorCall objectConstructor) {
+    Predicate<ExecutableSequence> outputTest;
+    if (GenInputsAbstract.dont_output_tests) {
+      outputTest = new AlwaysFalse<>();      
+    } else {
+      Predicate<ExecutableSequence> baseTest;
+      //base case: exclude sequences with just "new Object()", keep everything else
```

Comment:
I'm not really sure.  All of this is from previous code.  The Object constructor call *is* specifically added at the beginning, so it is probably an artifact of that. 

It seems like if being able to generate Object is an issue, it should go into the "seeds" instead.